### ISSUE TITLE
hasMany/hasOne not working with yidas/codeigniter-psr4-autoload

### ISSUE THREAD

**Issue by mikouaji** [yidas/codeigniter-model#12]
Hi, while using codeigniter-model with yidas/codeigniter-psr4-autoload a problem occurs.
When I create a hasMany relation in a model
```
public function users(){
return $this->hasMany('\app\models\User', 'group', 'id');
}
```
the codeigniter loader cannot find the class, if i change \app\models\User to User the loader cannot load the class because of the namespace.

For now I have found a solution to declare the model inside the function and it seems to solve the problem:
```
public function users(){
$u = new User();
return $this->hasMany('\app\models\User', 'group', 'id');
}
```
It took me a while so I'm posting to let you know about the problem, any maybe for anyone to find the solution.

**Comment by yidas** [comment]
Hi @mikouaji,

Are you using my PSR-4 autoload to define your model with namespace?

Regards
Nick

**Comment by mikouaji** [comment]
Hey, yes the models look something like this
```
namespace app\models;

use yidas\Model;

class User extends Model { ...
```

**Comment by yidas** [comment]
Hi @mikouaji,

It issue has been solved at [version 2.15.0](https://github.com/yidas/codeigniter-model/releases/tag/2.15.0).

BTW, in this version you can use static call just like Yii2 AR:

```php
$user = app\models\User::findOne(1);
```

Cheers
Nick Tsai

**Comment by mikouaji** [comment]
Hey, great job :)
